cloudify_azure/utils.py

```diff
@@ -457,8 +469,12 @@ def get_client_config(properties):
     if len(azure_config_keys) > len(client_config_keys):
         ctx.logger.warn("azure_config is deprecated please use client_config, "
                         "in later version it will be removed")
-        return azure_config
-    return client_config
+
+        final_client_config.update(azure_config)
```

Comment:
We should double check that we are not overwriting values with empty ones. For example, let's say we have this:

```
azure_config = {
  'key': None,
  'key2': 'foo'
}

client_config = {
  'key': 'foobar',
}

```

Then `{'key': 'foobar'}` will be overwritten by `{'key': None}`. This is not what we want.